src/status_im2/common/toasts/view.cljs

```diff
@@ -1,89 +1,56 @@
-                                                  evt-translation-y)))))
-              (gesture/on-end (fn [_]
-                                (when-not @dismissed-locally?
-                                  (reanimated/set-shared-value translate-y 0)
-                                  (create-timer)))))]
+      (let [duration     (or (rf/sub [:toasts/toast-cursor toast-id :duration]) 3000)
+            on-dismissed (or (rf/sub [:toasts/toast-cursor toast-id :on-dismissed]) identity)
+            create-timer #(reset! timer (background-timer/set-timeout close-toast duration))
+            translate-y  (reanimated/use-shared-value 0)
+            pan-gesture  (animation/pan-gesture {:clear-timer           clear-timer
+                                                 :create-timer          create-timer
+                                                 :translate-y           translate-y
+                                                 :close-toast           close-toast
+                                                 :set-dismissed-locally set-dismissed-locally
+                                                 :dismissed-locally?    @dismissed-locally?})]
         ;; create auto dismiss timer, clear timer when unmount or duration changed
         (rn/use-effect (fn [] (create-timer) clear-timer) [duration])
-        (rn/use-unmount #(on-dismissed id))
-        [gesture/gesture-detector {:gesture pan}
+        (rn/use-unmount #(on-dismissed toast-id))
+
+        [gesture/gesture-detector {:gesture pan-gesture}
          [reanimated/view
-          {;; TODO: this will enable layout animation at runtime and causing flicker on android
-           ;; we need to resolve this and re-enable layout animation
-           ;; issue at https://github.com/status-im/status-mobile/issues/14752
-           ;; :entering slide-in-up-animation
-           ;; :exiting  slide-out-up-animation
-           ;; :layout   reanimated/linear-transition
-           :style (reanimated/apply-animations-to-style
-                   {:transform [{:translateY translate-y}]}
-                   style/each-toast-container)}
-          [toast id]]]))))
+          {:entering animation/slide-in-up-animation
+           :exiting  animation/slide-out-up-animation
+           :layout   animation/linear-transition
+           :style    (reanimated/apply-animations-to-style
+                      {:transform [{:translateY translate-y}]}
+                      style/each-toast-container)}
+          [toast toast-id]]]))))
 
 (defn toasts
   []
-  (let [toasts-ordered (:ordered (rf/sub [:toasts]))]
-    [into
-     [rn/view
-      {:style style/outmost-transparent-container}]
-     (doall
-      (map (fn [id] ^{:key id} [:f> f-container id]) toasts-ordered))]))
+  (->> (rf/sub [:toasts])
+       :ordered
+       (into [rn/view {:style style/outmost-transparent-container}]
+             (map #(with-meta [:f> f-container %] {:key %})))))
```

Comment:
`into` here was called using `[ ]` instead of `( )`.

Also since we are using `into` we don't need that `doall` call for `map`.
so just refactored to use into and `map` in its xf form.

FYI, another possible refactor would be:

```clojure
[rn/view
 (map (fn [] ,,,)
      toast-ordered)]
```

we don't need `doall` unless we are using `deref (`@`) inside the `map` function. I'm using `into` since it was the intented function to be used.
 
```